### PR TITLE
Facing tooltip.

### PR DESCRIPTION
**Pull request by alexhliu**
Add a tooltip displaying the current camera facing direction.
image

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java [hunk 1]
```diff
@@ -2383,7 +2383,16 @@ public synchronized String sceneStatus() {
...
-        buf.append(String.format("pos: (%.1f, %.1f, %.1f)", pos.x, pos.y, pos.z));
+        buf.append(String.format("pos: (%.1f, %.1f, %.1f)\n", pos.x, pos.y, pos.z));
+
+        buf.append("facing: ");
+        double yaw = camera.getYaw();
+        yaw = (yaw + Math.PI*2) % (Math.PI*2);
+        int index = (int)Math.floor((yaw + Math.PI/8) / (Math.PI/4)) % 8;
+        buf.append(new String[]{"West", "SouthWest", "South", "SouthEast", "East", "NorthEast", "North", "NorthWest"}[index]);
+        index = (int)Math.floor((yaw + Math.PI/4) / (Math.PI/2)) % 4;
+        buf.append(new String[]{" (-X)", " (+Z)", " (+X)", " (-Z)"}[index]);
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:2394]:
```suggestion
        buf.append(" (towards ");
        buf.append(new String[]{"negative X", "positive Z", "positive X", "negative Z"}[index]);
        buf.append(")");
```

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java [hunk 2]
```diff
@@ -2383,7 +2383,16 @@ public synchronized String sceneStatus() {
...
         }
         Vector3 pos = camera.getPosition();
-        buf.append(String.format("pos: (%.1f, %.1f, %.1f)", pos.x, pos.y, pos.z));
+        buf.append(String.format("pos: (%.1f, %.1f, %.1f)\n", pos.x, pos.y, pos.z));
+
+        buf.append("facing: ");
+        double yaw = camera.getYaw();
+        yaw = (yaw + Math.PI*2) % (Math.PI*2);
+        int index = (int)Math.floor((yaw + Math.PI/8) / (Math.PI/4)) % 8;
+        buf.append(new String[]{"West", "SouthWest", "South", "SouthEast", "East", "NorthEast", "North", "NorthWest"}[index]);
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:2392]:
```suggestion
        buf.append(new String[]{"west", "southwest", "south", "southeast", "east", "northeast", "north", "northwest"}[index]);
```

**Review by leMaik — APPROVED** [review]
Looks great, I'd just like to adjust the wording to match Minecraft's debug menu (which should be well known by most users).

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java [hunk 3]
```diff
@@ -2383,7 +2383,16 @@ public synchronized String sceneStatus() {
           buf.append("\n");
         }
         Vector3 pos = camera.getPosition();
-        buf.append(String.format("pos: (%.1f, %.1f, %.1f)", pos.x, pos.y, pos.z));
+        buf.append(String.format("pos: (%.1f, %.1f, %.1f)\n", pos.x, pos.y, pos.z));
+
+        buf.append("facing: ");
+        double yaw = camera.getYaw();
+        yaw = (yaw + Math.PI*2) % (Math.PI*2);
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:2390]: What does this line do? Isn't it the same as `double yaw = camera.getYaw() % (Math.PI * 2)` ?

Comment (alexhliu) [reply chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:2390]: `camera.getYaw()` ranges from `-2PI` to `2PI`. This was just the simplest way I could think of converting it to `0` to `2PI`. A simple modulo would not convert the negative angles to positive.

Comment (leMaik) [reply chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:2390]: Unrelated to this PR but that range doesn't make any sense :smile: 0…2π would be enough

**Pull request merged**